dll/win32/shell32/utils.cpp

```diff
@@ -1294,3 +1294,30 @@ SHGetRealIDL(
 
     return hr;
 }
+
+static HRESULT
+GetCommandStringA(_In_ IContextMenu*pCM, _In_ UINT_PTR Id, _In_ UINT GCS, _Out_writes_(cchMax) LPSTR Buf, _In_ UINT cchMax)
+{
+    HRESULT hr = pCM->GetCommandString(Id, GCS & ~GCS_UNICODE, NULL, Buf, cchMax);
+    if (FAILED(hr))
+    {
+        WCHAR buf[MAX_PATH];
+        hr = pCM->GetCommandString(Id, GCS | GCS_UNICODE, NULL, (LPSTR)buf, _countof(buf));
+        if (SUCCEEDED(hr))
+            hr = SHUnicodeToAnsi(buf, Buf, cchMax) > 0 ? S_OK : E_FAIL;
+    }
+    return hr;
+}
+
+UINT
+GetDfmCmd(_In_ IContextMenu*pCM, _In_ LPCSTR verba)
```

Comment:
What DFM stands for?

```suggestion
GetDfmCmd(_In_ IContextMenu *pCM, _In_ LPCSTR verba)
```

Reply:
If that is "default menu", then `DefMenu` would be better.

We are an international team. Bad naming ruins us.

Reply:
> If that is "default menu", then `DefMenu` would be better.

If it was default, I would name it DefMenuCmd, not DfmCmd!

But it is not, it returns `DFM_CMD_*`.

Reply:
Thanks for explanation. Can you add a comment for it? Thanks.